q2_feature_table/_summarize/_visualizer.py

```diff
@@ -30,9 +30,12 @@
 TEMPLATES = pkg_resources.resource_filename('q2_feature_table', '_summarize')
 
 
-def tabulate_seqs(output_dir: str, data: DNAIterator) -> None:
+def tabulate_seqs(output_dir: str, data: DNAIterator,
+                  taxonomy: pd.DataFrame = None,
+                  metadata: qiime2.Metadata = None) -> None:
     sequences = []
     seq_lengths = []
+    metadata_df = metadata.to_dataframe()
```

Comment:
Need to guard against the possibility of metadata being None here. Maybe do the `to_dataframe` inside the if on line 57 since you don't appear to need the dataframe before that?